_projects/2023/100505771/100505771.Rmd

```diff
@@ -0,0 +1,284 @@
+    legend.key.width = unit(0.4, "cm"),
+    legend.title = element_text(color = "#616161", size = 6),
+    legend.justification = c(0.3, 0)
+  ) +
+  labs(title = " ", fill = paste("PRODUCT", "SECTORS", sep = "\n")) +
+  guides(fill = guide_legend(nrow = 1, override.aes = list(fill = new_order))) +
+  geom_text(
+    data = label_positions,
+    aes(x = Inf, y = Position, label = Sector),
+    color = "#E8E9EB",
+    angle = ifelse(label_positions$Sector == "Electronics", 15,
+                   ifelse(label_positions$Sector == "Minerals", 8,
+                          ifelse(label_positions$Sector == "Machinery", 8, 0))),
+    hjust = ifelse(label_positions$Sector == "Electronics", 2,
+                   ifelse(label_positions$Sector == "Machinery",  2.1,
+                          ifelse(label_positions$Sector == "Minerals", 5.2,
+                                 ifelse(label_positions$Sector == "Agriculture", 3, 1.8)))),
+    vjust = ifelse(label_positions$Sector == "Electronics", -2,
+                   ifelse(label_positions$Sector == "Machinery", -.9,
+                          ifelse(label_positions$Sector == "Minerals", -5,
+                                 ifelse(label_positions$Sector == "Agriculture", 0.6, 0.2)))),
+    size = ifelse(label_positions$Sector == "Electronics", 4,
+                  ifelse(label_positions$Sector == "Machinery", 4,
+                         ifelse(label_positions$Sector == "Minerals", 3,
+                                ifelse(label_positions$Sector == "Agriculture", 2.8, 6)))),
+    na.rm = TRUE
+  )
+
+
+
+```
+
+There were some series of challenged that added both complexity and valuable lessons, to start with, the first challenge was on the current gross export amounts, since the original data set already had the sign "\$" and the letter "B" on the numbers, the ggplot didn't organize well the amounts, I had to remove those characters from the "Current Gross Export" variable and then convert it to number as below,
+
+```{r}
+
+exportmex <- exportmex |> mutate(`Current Gross Export`= sub("\\$", "",`Current Gross Export`)) 
+Current_Gross_Export <- as.numeric(exportmex$`Current Gross Export`)
+
+```
+
+In second place, I must mention the sectors, since ggplot2 order alphabetically by default I needed to modify them so I can give them the same order as on the original graph.
+
+```{r}
+colors <- c("#b33d6d","#2b607c","#7cdada","#7ba2d9","#8c7ad7","#c47cda","#d87b7a","#bb9689","#dab57c","#f4d025","#7cdaa0")
+
+```
+
+An aesthetic where I invested a lot of time was, setting up the correct angles and direction of each label per sector, I used if else function so I can move the label of each sector separately, I had to be very meticulous on this because if I move the angle the other aesthetics of the label would change, and to set them up exactly as the original plot was a challenge.
+
+```{r}
+geom_text(
+    data = label_positions,
+    aes(x = Inf, y = Position, label = Sector),
```

Comment:
If you want just to show a piece of code without executing it, add `eval=FALSE` in the chunk options.